app/events/views.py

```diff
@@ -150,22 +175,36 @@ def get():
             user_id = User.decode_token(access_token)
             if not isinstance(user_id, str):
                 # get all the events for this user
+                categories = EventCategory.get__all_categories()
+                page = request.args.get('page', default=1, type=int)
+                limit = request.args.get('limit', default=10, type=int)
+
                 events = Events.get_all_user(user_id)
+
+                event_page = events.paginate(page, limit, False).items
                 results = []
 
-                for event in events:
+                for event in event_page:
+                    for category in categories:
+                        event.event_category = category.category_name
```

Comment:
are you re-assigning the `event_category` attribute to a random category in this line?

Reply:
this is a function am yet to finish, was testing it out.